app/serializers/search_form_serializer.rb

```diff
@@ -15,22 +15,37 @@ def sort
 
   def query
     {
-      'bool' => {
-        'must': [
-          { 'match': { 'postcode_searchable': true } },
-          {
-            'nested' => {
-              'path' => 'advisers',
-              'filter' => {
-                'geo_distance' => {
-                  'distance' => '650miles',
-                  'location' => object.coordinates.reverse
-                }
-              }
+      'bool': {
+        'must': (build_postcode_filters + build_types_of_advice_filters)
+      }
+    }
+  end
+
+  private
+
+  def build_postcode_filters
+    [
+      {'match': {'postcode_searchable': true }},
+      {
+        'nested' => {
+          'path' => 'advisers',
+          'filter' => {
+            'geo_distance' => {
+              'distance' => '650miles',
+              'location' => object.coordinates.reverse
             }
           }
-        ]
+        }
       }
-    }
+    ]
+  end
+
+  def build_types_of_advice_filters
+    [].tap do |filters|
+      filters << { match: { 'options_when_paying_for_care': true } } if object.options_when_paying_for_care == '1'
```

Comment:
Hows about something like:

```ruby
%i(options_when_paying_for_care equity_release inheritance_tax_planning wills_and_probate).map do |f|
    { match: { f => true } } if object.public_send(f) == '1'
end
```

The serializer shouldn't have to know about the `== '1'` predicate either. We can do something nice in the `SearchForm` in this regard too. Give me a shout and we'll sit down together.